### DISCUSSION TITLE
Select police or auth.uid() not working as expected?

### DISCUSSION THREAD

**Discussion by oscarrc** [supabase/supabase#4049] [Questions] [answered]
Hello everyone. I'm new here, just trying Supabase but i've run into a problem.

I'm triying to only allow to read a row to the user that created it. For that, i've an user_id field in the table and a policy for select using `auth.uid() = lists.user_id`

![image](https://user-images.githubusercontent.com/7826103/143411320-fcc52c94-a409-4c1c-b720-273336e632f1.png)

The problem is the query `supabase.from('lists').select('*')` to that table returns an empty array, however:

* If I  remove the policy and query `supabase.from('lists').select('*').eq('user_id', 'the actual user uuid')` it works
* If I hardcode de user uuid into the policy (` 'actual user id':: uuid = lists.user_id` ) it works too

This is my only policy on that table at the moment, and I'm creating a row on that list at the moment of signup. 

I don't really know what I'm doing wrong because according to the docs, this is what I've to do. Maybe auth.uid() isn't returning user's uuid or something; because if it were it shoud work, shouldn't it?.  

Any help will be appreciated.

**Comment by GaryAustin1** [comment, marked as answer]
I’d first double check the user is actually signed in when you do the select.

**Reply by oscarrc** [reply]
As the saying goes, two heads are better than one.

The user was logged in but I've trying to select the data server side on Next.js and there was no token in the request. I've solved it with [this](https://github.com/supabase/supabase/issues/1735#issuecomment-922284089)

Dumb, dumb me
